src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -305,13 +305,23 @@ internal JsonTypeInfo(Type type, JsonConverter converter, Type runtimeType, Json
                     {
                         ElementType = converter.ElementType;
                         CreateObject = Options.MemberAccessorStrategy.CreateConstructor(runtimeType);
+
+                        if (converter.RequiresDynamicMemberAccessors)
+                        {
+                            converter.Initialize(Options, this);
```

Comment:
does this account for the possibility that one converter instance is initialized by multiple `JsonTypeInfo` instances? I _think_ that it's occassionally possible although not sure it applies to converters that override `RequiresDynamicMemberAccessors`.

Reply:
Hmm I think it could happen occasionally due to threading, but I think even in that case each `JsonTypeInfo` construction would create a separate converter instance, and the first one to complete would win, discarding the duplicates. Each converter instance would only be for a specific type, so there are no issues of creating invalid dynamic accessors.